src/tests/test_magic.py

```diff
@@ -196,6 +196,193 @@ def test_persist_bare(ip):
+                )
+            ],
+        ),
+        ("number_table", [(0, 4, -2)]),
+    ],
+)
+def test_persist_replace_override(ip, test_table, expected_result):
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    ip.run_cell(f"%sql --persist sqlite:// {saved_df_name}")
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+
+    out = ip.run_cell(f"%sql SELECT * FROM {saved_df_name}")
+    # To test it's referring to new saved dataframe
+    assert out.result == expected_result
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "test_table", [("test"), ("author"), ("website"), ("number_table")]
+)
+def test_persist_replace_override_reverted_order(ip, test_table):
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    out = ip.run_cell(f"%sql --persist sqlite:// {saved_df_name}")
+
+    assert "Table already exists; consider using --persist-replace." in str(
+        out.error_in_exec
+    )
+
+
+@pytest.mark.parametrize(
+    "test_table", [("test"), ("author"), ("website"), ("number_table")]
+)
+def test_persist_and_append_use_together(ip, test_table):
+    # Test error message when use --persist and --append together
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    out = ip.run_cell(f"%sql --persist-replace --append sqlite:// {saved_df_name}")
+
+    assert """You cannot simultaneously replace and append data to a dataframe;
+                  please choose to utilize either one or the other.""" in str(
+        out.error_in_exec
+    )
+    assert (out.error_in_exec.error_type) == "ValueError"
+
+
+@pytest.mark.parametrize(
+    "test_table, expected_result",
+    [
+        ("test", [(0, 1, "foo")]),
+        ("author", [(0, "William", "Shakespeare", 1616)]),
+        (
+            "website",
+            [
+                (
+                    0,
+                    "Bertold Brecht",
+                    "https://en.wikipedia.org/wiki/Bertolt_Brecht",
+                    1954,
+                )
+            ],
+        ),
+        ("number_table", [(0, 4, -2)]),
+    ],
+)
+def test_persist_and_persist_replace_use_together(
+    ip, capsys, test_table, expected_result
+):
+    # Test error message when use --persist and --persist-replace together
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    ip.run_cell(f"%sql --persist --persist-replace sqlite:// {saved_df_name}")
+    out, _ = capsys.readouterr()
+    execute_out = ip.run_cell(f"%sql SELECT * FROM {saved_df_name}")
+
+    # Test warning message
+    assert "Please use either --persist or --persist-replace" in out
+    # Test persist-replace is used
+    assert execute_out.result == expected_result
```

Comment:
Same as before, we need another table (different) to make sure the relevant data is used.